### PR TITLE
`<ColorInput/>` -  add `preset` & `showClear` props

### PR DESCRIPTION
**Pull request by MSiliunas**
### 🔦 Summary
- Added support for no color selection in `<ColorInput />`;
- Added `predefinedColors` to `<ColorPicker />`;

### ✅ Checklist
- [x] 👨‍💻 API change is approved by the UI Infra Developers 
- [x] 👨‍🎨 UX change is approved by the Design System UX 
- 📚 Change is documented
  - [x] Story
  - [x] API description
- 🔬 Change is tested
  - [x] Component

### CODE REVIEW

File: src/ColorInput/ColorInput.private.uni.driver.js [hunk 1]
```diff
@@ -24,5 +26,8 @@ export const colorInputPrivateDriverFactory = base => {
     isViewerNull: async () =>
       await base.$('[data-hook="colorinput-viewer-line"]').exists(),
     getViewerSize,
+    /** Return Swatches's component testkit methods */
+    swatchesDriver: async () =>
```
Comment (argshook) [inline src/ColorInput/ColorInput.private.uni.driver.js]: you probably already know, but just for the record, we don't advise to expose drivers within other drivers. Use only specific methods that you need

File: src/ColorInput/components/ColorViewer.js [hunk 2]
```diff
@@ -1,24 +1,28 @@
 import React from 'react';
-import Color from 'color';
 
 import Popover from '../../Popover';
 import ColorPicker from '../../ColorPicker';
 import styles from './ColorViewer.st.css';
 
 export class ColorViewer extends React.Component {
+  onChange = _color => {
+    this.props.onChange(_color.alpha() === 0 ? '' : _color.hex());
```
Comment (argshook) [inline src/ColorInput/components/ColorViewer.js]: are you sure you'll always get a color object with these methods? previously it was guarded with `Color`, maybe for a reason?

Comment (MSiliunas) [reply src/ColorInput/components/ColorViewer.js]: Added the `Color` wrap again 👍

File: src/ColorPicker/ColorPicker.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 .root {
-  width: 230px;
+  width: 240px;
```
Comment (argshook) [inline src/ColorPicker/ColorPicker.scss]: @MSiliunas this is according to the spec, right?

Comment (MSiliunas) [reply src/ColorPicker/ColorPicker.scss]: yes, 
image

**Comment by milkyfruit** [comment]
### `<Swatches />`
Issues:
- [ ] `size` should not exist, instead component should have `columns` prop.

Missing props:
- [ ] Add color button
- [ ] Remove color button
- [ ] Title
- [ ] Maximum number of colors

### `<ColorPicker />`

Issues:
- [ ] Swatches area paddings should be `12px 18px` If swatches contain small title, then paddings are `9px 18px 12px 9px`. By design it's handled in `<Swatches />` component, but if it makes more sense to apply it in ColorPicker, let me know.
- [ ] Footer area paddings (where x,v actions are) should be `18px`.
- [ ] Input area paddings should be `12px 18px`
- [ ] Missing color previewer next to the input.

@MSiliunas @argshook

**Comment by MSiliunas** [comment]
@milkyfruit https://github.com/wix/wix-style-react/pull/3499

>  ~Swatches area paddings should be 12px 18px~ If swatches contain small title, then paddings are 9px 18px 12px 9px. By design it's handled in <Swatches /> component, but if it makes more sense to apply it in ColorPicker, let me know.
  ~Footer area paddings (where x,v actions are) should be 18px.~
  ~Input area paddings should be 12px 18px~
  ~Maker cancel / Confirm button size small~

**Pull request merged**